test/prototype_common_utils.py

```diff
@@ -64,36 +241,60 @@ def make_image(size=None, *, color_space, extra_dims=(), dtype=torch.float32, co
             features.ColorSpace.RGB_ALPHA: 4,
         }[color_space]
     except KeyError as error:
-        raise pytest.UsageError() from error
+        raise pytest.UsageError(f"Can't determine the number of channels for color space {color_space}") from error
+
+    def fn(shape, dtype, device):
+        max_value = get_max_value(dtype)
+        data = torch.testing.make_tensor(shape, low=0, high=max_value, dtype=dtype, device=device)
+        if color_space in {features.ColorSpace.GRAY_ALPHA, features.ColorSpace.RGB_ALPHA} and constant_alpha:
+            data[..., -1, :, :] = max_value
+        return features.Image(data, color_space=color_space)
 
-    shape = (*extra_dims, num_channels, *size)
-    max_value = get_max_value(dtype)
-    data = make_tensor(shape, low=0, high=max_value, dtype=dtype)
-    if color_space in {features.ColorSpace.GRAY_ALPHA, features.ColorSpace.RGB_ALPHA} and constant_alpha:
-        data[..., -1, :, :] = max_value
-    return features.Image(data, color_space=color_space)
+    return ImageLoader(fn, shape=(*extra_dims, num_channels, *size), dtype=dtype, color_space=color_space)
 
 
-make_grayscale_image = functools.partial(make_image, color_space=features.ColorSpace.GRAY)
-make_rgb_image = functools.partial(make_image, color_space=features.ColorSpace.RGB)
+make_image = from_loader(make_image_loader)
 
 
-def make_images(
-    sizes=((16, 16), (7, 33), (31, 9)),
+def make_image_loaders(
+    *,
+    sizes=DEFAULT_IMAGE_SIZES,
     color_spaces=(
         features.ColorSpace.GRAY,
         features.ColorSpace.GRAY_ALPHA,
         features.ColorSpace.RGB,
         features.ColorSpace.RGB_ALPHA,
     ),
+    extra_dims=DEFAULT_EXTRA_DIMS,
     dtypes=(torch.float32, torch.uint8),
-    extra_dims=((), (0,), (4,), (2, 3), (5, 0), (0, 5)),
+    constant_alpha=True,
 ):
     for size, color_space, dtype in itertools.product(sizes, color_spaces, dtypes):
-        yield make_image(size, color_space=color_space, dtype=dtype)
+        yield make_image_loader(size, color_space=color_space, dtype=dtype, constant_alpha=constant_alpha)
```

Comment:
These functions don't need to yield since we don't load the tensors anyway. Still, I prefer this idiom over the more verbose

```py
loaders = []

for ...:
    loaders.append(...)

for ...:
    loaders.append(...)

return loaders
```

Since we only iterate over the returned values this makes no difference.